internal/host/validator.go

```diff
@@ -1229,18 +1229,179 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
 	}
 }
 
-func shouldValidateDnsResolution(c *validationContext) bool {
-	// Skip DNS resolution checks in IPI network mode
-	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+func ignitionHasFile(ignition *ignition_types.Config, path string) bool {
+	for _, file := range ignition.Storage.Files {
+		if file.Path == path {
+			return true
+		}
+	}
+
+	return false
+}
+
+func ignitionContainsManagedNetworkingFiles(config *ignition_types.Config) bool {
+	if ignitionHasFile(config, "/etc/kubernetes/manifests/coredns.yaml") ||
+		ignitionHasFile(config, "/etc/kubernetes/manifests/keepalived.yaml") {
+		// This is a hack - since we have no official way to know whether a
+		// worker ignition file originated from a cluster with managed
+		// networking or not, we instead rely on the presence of coredns and
+		// keepalived pod manifests to indicate that. We only expect those to
+		// be present in clusters with managed networking. To be a bit more
+		// robust, we consider the presence of any one of them to mean that the
+		// cluster has managed networking. This gives us better forwards
+		// compatibility if either one of them gets renamed / replaced with
+		// other technologies in future OCP versions.
+		//
+		// Another way in which this is hacky is that users could manually
+		// create static pods with the same name as part of their
+		// machine-configs, in which case we would have a false-positive
+		// detection. But that is admittedly very unlikely.
+		//
+		// Hopefully we can negotiate with the relevant OCP teams to have a
+		// more official, stable way to have this detection - like a magic
+		// empty file placed somewhere in the ignition that we can check for
+		// the presence of. Once we have such file, we can slowly deprecate
+		// this detection mechanism and fully move to the new one by inspecting
+		// that file instead.
+		return true
+	}
+
+	return false
+}
+
+// importedClusterHasManagedNetworking checks imported clusters in order to
+// determine whether they have managed networking or not. It does so by
+// inspecting the ignition file returned by each of the day-2 host agent's API
+// connectivity step responses and looking for file entries within them which
+// indicate that the cluster they originated from has managed networking.
+func (v *validator) importedClusterHasManagedNetworking(cluster *common.Cluster) bool {
+	for _, day2Host := range cluster.Hosts {
+		if day2Host.APIVipConnectivity == "" {
+			continue
+		}
+
+		var apiConnectivityResponse models.APIVipConnectivityResponse
+		if err := json.Unmarshal([]byte(day2Host.APIVipConnectivity), &apiConnectivityResponse); err != nil {
+			v.log.WithError(err).Warnf("Invalid API connectivity response")
+			continue
+		}
+
+		if !apiConnectivityResponse.IsSuccess {
+			continue
+		}
+
+		config, _, err := v3_2.Parse([]byte(apiConnectivityResponse.Ignition))
```

Comment:
will it be always 3.2 ?

Reply:
Probably not, but I think v3_2.Parse will handle any 3.x versions as they're supposed to be backwards compatible. 

@carbonin what would be the correct way to behave here? Parsing worker.ign ignition files from arbitrary clusters the user want us to help them join

Reply:
I think you might want something more like [`ParseCompatibleVersion`](https://github.com/coreos/ignition/blob/d508eb2e63df37d0069d93ef85cfafd548b6c63e/config/v3_2/config.go#L64). I'm not sure if the current implementation would error for 3.1, but it's probably worth a unit test.

Reply:
And what about forwards compatibility with future ignition versions?